presto-main/src/main/java/io/prestosql/execution/QueryStateTimer.java

```diff
@@ -87,14 +93,20 @@ private void beginDispatching(long now)
 
     public void beginPlanning()
     {
-        beginPlanning(tickerNanos());
+        beginPlanning(tickerNanos(), false);
     }
 
-    private void beginPlanning(long now)
+    // QueryStateTimer will still time the planning time stats even a planning is not involved
+    // (e.g. DDL statements don’t go through planning) in this case the planningCpuTime should be 0
+    // and beginPlanningCpuNanos should be unset, isCascadeRecord is the flag to help differentiate
+    private void beginPlanning(long now, boolean isCascadeRecord)
```

Comment:
let's make it consistent. If `planningTime` is considered for DDL, then let's also account for CPU for consistency

Reply:
Updated, please see new comments.